--- origsrc/shared-mime-info-0.70/Makefile.am
+++ src/shared-mime-info-0.70/Makefile.am
@@ -67,10 +67,10 @@ uninstall-hook:
 	rm -f "$(DESTDIR)$(datadir)/mime/treemagic"
 	rm -f "$(DESTDIR)$(datadir)/mime/types"
 
-all: check
+all: check-xml
 	$(MAKE) -C po shared-mime-info.pot
 
-check: freedesktop.org.xml check-translations
+check-xml: freedesktop.org.xml check-translations
 	if test -n $(XMLLINT) ; then \
 		xmllint --noout --valid $(top_srcdir)/freedesktop.org.xml;	\
 	fi
