diff -ur origsrc/shared-mime-info-2.0/data/freedesktop.org.xml.in src/shared-mime-info-2.0/data/freedesktop.org.xml.in
--- origsrc/shared-mime-info-2.0/data/freedesktop.org.xml.in
+++ src/shared-mime-info-2.0/data/freedesktop.org.xml.in
@@ -2991,6 +2991,7 @@
   </mime-type>
   <mime-type type="application/x-ms-dos-executable">
     <comment>DOS/Windows executable</comment>
+    <sub-class-of type="application/x-executable"/>
     <generic-icon name="application-x-executable"/>
     <magic priority="50">
       <match type="string" value="MZ" offset="0"/>
@@ -3556,6 +3557,9 @@
       <match type="string" mask="0xffffffff000000000000000000000000ff" value="\177ELF            \003" offset="0"/>
     </magic>
     <glob pattern="*.so"/>
+    <glob pattern="*.dll"/>
+    <glob pattern="*.cmxs"/>
+    <glob pattern="*.oct"/>
   </mime-type>
   <mime-type type="application/x-shellscript">
     <comment>shell script</comment>
