--- origsrc/shared-mime-info-2.4/data/freedesktop.org.xml.in
+++ src/shared-mime-info-2.4/data/freedesktop.org.xml.in
@@ -3959,6 +3959,9 @@
     </magic>
     <glob pattern="*.so"/>
     <glob weight="60" pattern="*.so.[0-9]*"/>
+    <glob pattern="*.dll"/>
+    <glob pattern="*.cmxs"/>
+    <glob pattern="*.oct"/>
   </mime-type>
   <mime-type type="application/x-shellscript">
     <comment>Shell script</comment>
