--- origsrc/shared-mime-info-0.40/Makefile.am	2008-06-11 13:24:50.000000000 -0500
+++ src/shared-mime-info-0.40/Makefile.am	2008-07-07 00:17:01.171875000 -0500
@@ -60,7 +60,6 @@
 	rm -f "$(DESTDIR)$(datadir)/mime/generic-icons"
 	rm -f "$(DESTDIR)$(datadir)/mime/globs2"
 
-all: check
 
 check:
 	@if test -n $(XMLLINT) ; then \
--- origsrc/shared-mime-info-0.40/update-mime-database.c	2008-06-09 04:09:49.000000000 -0500
+++ src/shared-mime-info-0.40/update-mime-database.c	2008-07-07 00:17:01.203125000 -0500
@@ -55,6 +55,7 @@
 	"multipart",
 	"x-content",
 	"x-epoc",
+	"chemical",
 };
 
 /* Represents a MIME type */
